### PR TITLE
add Miri error variant for process exit

### PR DESCRIPTION
**Pull request by RalfJung**
This is to support https://github.com/rust-lang/miri/pull/702

r? @oli-obk

### CODE REVIEW

**Review by bjorn3 — APPROVED** [review]

File: src/librustc/mir/interpret/error.rs [hunk 1]
```diff
@@ -229,6 +229,10 @@ pub enum InterpError<'tcx, O> {
     /// match an existing variant.
     MachineError(String),
 
+    /// Not actually an interprer error -- used to signal that execution has exited
```
Comment (oli-obk) [inline src/librustc/mir/interpret/error.rs]:
```suggestion
    /// Not actually an interpreter error -- used to signal that execution has exited
```

File: src/librustc/mir/interpret/error.rs [hunk 2]
```diff
@@ -229,6 +229,10 @@ pub enum InterpError<'tcx, O> {
...
+    /// with the given status code.
```
Comment (oli-obk) [inline src/librustc/mir/interpret/error.rs]: Mention that this is a miri thing and not a const eval thing.

**Review by oli-obk — CHANGES_REQUESTED** [review]
r=me with doc nits

**Pull request merged**